app/Controllers/BaseController.php

```diff
@@ -2,18 +2,58 @@
 
 namespace App\Controllers;
 
+use App\Models\User;
+use Core\Constants\Constants;
+use Lib\FlashMessage;
+
 class BaseController
 {
+    protected ?User $currentUser = null;
+
+    protected function currentUser(): ?User
+    {
+        if ($this->currentUser === null && isset($_SESSION['user']['id'])) {
+            $this->currentUser = User::find($_SESSION['user']['id']);
+        }
+        return $this->currentUser;
+    }
+
+    protected function authenticated(): void
+    {
+        if ($this->currentUser() === null) {
+            FlashMessage::danger('Você deve estar logado para acessar essa página.');
+            $this->redirectToRoute('login.form');
+            exit;
+        }
+    }
+
+    protected function isAdmin(): bool
+    {
+        $user = $this->currentUser();
+        return $user && $user->getRole() === 'admin';
```

Comment:
Better option:

`$this->currentUser()->isAdmin()`
`$this->currentUser()->is('admin')`